### ISSUE TITLE
Add operators for 'you'd be/have/do' and 'we'd be/have/do' tags

### ISSUE THREAD

**Issue by fallerOfFalls** [AndreaUnlimited/starfriends#9]
Suggestions:

```
- NOT

| OR

: BETWEEN
```

**Comment by fallax** [comment]
I was half way through writing an issue for exactly this. At the very least, we need some way of making it clear how putting down multiple contradictory things (e.g. I am happy to be in a relationship with either a non-smoker or a smoker, so I put down "smoker non-smoker") affects match scores.

**Comment by fallerOfFalls** [comment]
If you're happy with either then I'm not sure why you'd tag it at all. But yes, you could do smoker|non-smoker and then your match score with "people who specify whether they smoke" would increase relative to people who don't specify.

**Comment by fallax** [comment]
@fallerOfFalls I was thinking about things like "I am interested in men and non-binary people but not women"; as you say, "NOT" is more useful for that. I think "NOT" is the real use case for operators - e.g. "I am looking for NOT straight"

**Comment by fallerOfFalls** [comment]
Ah I see (I think), if you're interested in men and non-binary people but not women, you might put `man|nonbinary! -woman!` (using the REQUIRED operator from #11). Looking for NOT straight would be `-straight` if it's a preference or `-straight!` if you absolutely don't even want to see straight people.

**Comment by fallerOfFalls** [comment]
So, as well as NOT, OR and BETWEEN, we can have:

```
! REQUIRED
```

as a suffix - see #11;
and

```
* WILDCARD
```

 just an idea. dunno how that would work though.

**Comment by Cassolotl** [comment]
having a required field is going to be much easier for some people though.

Perhaps it could be a setting: operators, or separate tag sections?

when i say some people, i think probably most people

most people don't want to have to learn a snippet of programming thought process to use an online dating site

operators as default could be a significant obstacle

I'd find it fun but i can imagine that most people won't

(Copied from the telegram group, and originally written by me.)

**Comment by fallerOfFalls** [comment]
Yes, separate fields could be an input option, and not just for the required operator, but also NOT for example. It sounds way more complicated to me, but I think it could end up something like when you do a google advanced search and it shows you a load of fields and then when you click search it uses what you put in the fields to assemble a single field search using operators. Yes, good.

**Comment by fallerOfFalls** [comment]
Another idea about this: in addition to symbol operators like `- | : ! *` word operators `NOT OR BETWEEN REQUIRED ANYTHING` could work. word operators would be easier for some people to use/remember, while symbol operators mean less typing. so why not have both?

Also I think it should be done so that it doesn't matter whether there's a space in between the operator and the argument it's 'operating' on.

**Comment by Cassolotl** [comment]
[There's a bounty on this](https://docs.google.com/spreadsheets/d/1qT56eFRbk5klbJ29k8Zf7eRhdTkw1awF4YGQM6cW-f8/edit?usp=sharing). Email me (Cassian) as per instructions in the bounty spreadsheet to add your bounty!

**Comment by Cassolotl** [comment]
There is now a NOT operator: !tag

It can be used in "I am", "you'd be" and "we'd be".

**Comment by Cassolotl** [comment]
While I'm here tidying up, I'll add that a Required operator is in the works. :)

**Comment by Cassolotl** [comment]
The Required operator is on the list for Beta. :)
